clients/cloud/csharp/Program.cs

```diff
@@ -37,16 +37,8 @@ static async Task<ClientConfig> LoadConfig(string configPath, string certDir)
                     .ToDictionary(
                         line => line.Substring(0, line.IndexOf('=')),
                         line => line.Substring(line.IndexOf('=') + 1));
-                Enum.TryParse(cloudConfig["sasl.mechanisms"], out SaslMechanism saslMechanism);
-                Enum.TryParse(cloudConfig["security.protocol"], out SecurityProtocol securityProtocol);
-                var clientConfig = new ClientConfig
-                {
-                    BootstrapServers = cloudConfig["bootstrap.servers"].Replace("\\", ""),
-                    SaslMechanism = saslMechanism,
-                    SecurityProtocol = securityProtocol,
-                    SaslUsername = cloudConfig["sasl.username"],
-                    SaslPassword = cloudConfig["sasl.password"],
-                };
+
+                var clientConfig = new ClientConfig(cloudConfig);

```

Comment:
ok, this is probably best. but some commentary:

fwiw, the issue is the sasl protocol name has underscores in it in the config file, e.g. `sasl_ssl`, but this doesn't follow C# naming conventions so our enum values don't follow the standard config value names (they remove the underscore), so parsing the enum string doesn't work. the required translation is implemented in the `Config` class, and if you use the `Set` method (or constructor as you're doing), it'll all just work. which is great, but it's not ideal for the example as it's complicated to explain, and we don't want to be highlighting features that people shouldn't be using, which we are here.

the root cause of my concerns with all this is the config file is in an inappropriate format for the C# examples.

anyway, we do demonstrate normal use with the `ProducerConfig` and `ConsumerConfig` later on, so i guess that's all good. still not happy about this though.

one more thing: using `Enum.TryParse` isn't appropriate here - it's intended to use in `if` statements, the try parse will just fail silently without looking at the return value.